### ISSUE TITLE
Get metadata for all DR9 quasars

### ISSUE THREAD

**Issue by dkirkby** [belaa/boss-pipeline-errors#1]
I don't think this has been tested yet, but should be possible using something like `--quasar-catalog-name=DR9Q` with `bossquery`.

**Comment by dkirkby** [comment]
The reason we are starting with DR9 is to reduce the data volume (hopefully all the (non-lite) spec files will fit on your laptop) and because there are published plots that you can compare against.

The DR9Q catalog is documented [here](https://www.sdss3.org/dr9/algorithms/qso_catalog.php) and in this [paper](http://arxiv.org/abs/1210.5166).

As a check of your catalog selection, try to reproduce the following plots from the paper:
- PLUG_RA, PLUG_DEC distribution in Fig.1
- Integrated sample size in Fig.2
- r-PSF magnitude in Fig.6 (I'm not sure which spAll variable corresponds to r-PSF)
- Redshift distribution (Z_VI ?) in black histogram of Fig.23

**Comment by dkirkby** [comment]
Don't forget to include `--quasar-catalog` at the same time as the `--quasar-catalog-name DR9Q` option!

**Comment by belaa** [comment]
Here are the first two plots from the paper. I downloaded confirmed DR9 quasars using the following options (PLUG_RA and PLUG_DEC were invalid columns in this specific data model): 

`--what PLATE,MJD,FIBER,RA,DEC,Z_VI --where 'Z_VI>2.1 AND Z_VI<5.5 AND ZWARNING==0' --quasar-catalog --quasar-catolog-name DR9Q`

The number of quasars in my data file differed from the number in Daniel's [here](https://trac.sdss3.org/attachment/wiki/BOSS/LyaForestsurvey/PipelineErrors/dr9-qso-hiz.dat). I had a total of 61849 quasars, whereas he had 67906. In Fig. 2, I was able to see the shutdown period during the summer of 2010 at around MJD=55400. 

Fig. 1:
![dr9_space_dist](https://cloud.githubusercontent.com/assets/9096399/8609023/b1aa4f7e-2655-11e5-9605-8d2fb31b50b0.png)

Fig.2:
![cumul_quasars](https://cloud.githubusercontent.com/assets/9096399/8609075/1739676c-2656-11e5-9349-7c43e8e6e4a9.png)

**Comment by belaa** [comment]
I was having some trouble figuring out what variable corresponds to r-PSF. Does it refer to the magnitude in the 'r' filter? Would RFLUX (r-band flux) be the appropriate variable?

**Comment by dkirkby** [comment]
Yes, its the "PSF magnitude" over the range of wavelengths covered by the r-band filter (http://classic.sdss.org/dr7/instruments/imager/#filters).  The PSFMAG_0...4 variables from `bossquery` give the PSF magnitudes for each filter (0=u, 1=g, 2=r, 3=i, 4=z), so you should be able to add `PSFMAG_2` to the `--where` parameter to include a column of these values.

**Comment by belaa** [comment]
I was able to get the r-PSF values for DR12Q, but the PSFMAG column does not exist in the data model for [DR9Q](http://data.sdss3.org/datamodel/files/BOSS_QSO/DR9Q/DR9Q.html). However, [this](https://www.sdss3.org/dr9/algorithms/magnitudes.php#mag_psf) page, which describes Data Release 9, states that psfMag is in fact accessible. I'm going to ask Daniel about this today, but in the meantime I'm going to just specify `--where 'MJD<=55773'` in the DR12Q catalog to include all quasars recorded up to July 2011 (end of DR9) so that I can access these values.

**Comment by belaa** [comment]
Here is Fig. 23 for the full DR9 sample: 

![quasars_vs_redshift](https://cloud.githubusercontent.com/assets/9096399/8663805/52db8044-2984-11e5-8491-36e4474c7b0f.png)

**Comment by belaa** [comment]
And Fig. 6: 

![rpsf](https://cloud.githubusercontent.com/assets/9096399/8666037/28f5da9c-29a0-11e5-8825-5bbb1eec9846.png)

Although my graph is normalized differently...The y-axis in the paper ranges from 0.00-0.08.

**Comment by dkirkby** [comment]
It looks like the difference in normalization is equal to the bin width, i.e., you have normalized so that your vertical scale is "quasars per unit of magnitude" while the paper uses a vertical scale of "quasars per histogram bin".  Both are fine as long as the axis label makes it clear which was used.
